### PR TITLE
t/67: Moved wireframe CSS to a separate theme.

### LINKED ISSUE #67: Extract wireframe to a separate theme
ATM, it's hard to develop an application based on CKEditor's UI framework which is to be visually distinctive from Lark. Developers are forced to override Lark's CSS, which produces a tangled mess of specificity problems etc.

So to bypass the problem, a new wireframe theme is to be created, which will define only the basics and let any theme built upon it to define own styling.

### PR DESCRIPTION
**Pull request by oleq**
Closes #67.

### CODE REVIEW

File: theme/components/button.scss [hunk 1]
```diff
@@ -71,13 +46,21 @@
...
-	@include ck-unselectable();
-	@include ck-rounded-corners();
-	@include ck-tooltip_enabled();
+	&:focus {
+		@include ck-focus-ring();
+		@include ck-box-shadow( $ck-focus-outer-shadow );
+	}
 
 	@include ck-button-icon {
 		float: left;
```
Comment (oleq) [inline theme/components/button.scss]: Shouldn't floats be managed by the Wireframe? I'm not sure about it.

File: theme/components/dropdown.scss [hunk 2]
```diff
@@ -2,23 +2,14 @@
...
-		content: '';
-		width: 0;
-		height: 0;
 		border-style: solid;
 		border-width: .4em .4em 0 .4em;
 		border-color: ck-color( 'text', 30 ) transparent;
-		pointer-events: none;
 
-		position: absolute;
 		top: 50%;
```
Comment (oleq) [inline theme/components/dropdown.scss]: It's pretty obvious that the dropdown "triangle" should be in the middle of the dropdown. Shouldn't it land in the Wireframe?

File: theme/components/dropdown.scss [hunk 3]
```diff
@@ -2,23 +2,14 @@
...
 		right: ck-spacing();
-		z-index: ck-z();
 		transform: translate( 0, -50% );
```
Comment (oleq) [inline theme/components/dropdown.scss]: Same as above.

File: theme/components/dropdown.scss [hunk 4]
```diff
@@ -36,26 +27,15 @@
...
 
 .ck-dropdown__panel {
-	// This is to get rid of flickering when the tooltip is shown under the panel,
-	// which looks like the panel moves vertically a pixel down and up.
-	-webkit-backface-visibility: hidden;
-
 	border: 1px solid ck-border-color( 'foreground' );
-	display: none;
-	position: absolute;
 	left: 0px;
```
Comment (oleq) [inline theme/components/dropdown.scss]: Shouldn't it also land in the Wireframe? OTOH, there could be a theme which positions the panel in a different way. Hard to tell.

File: theme/components/dropdown.scss [hunk 5]
```diff
@@ -36,26 +27,15 @@
...
 
 	// Compensate double border from button and from box when positioned below the button.
 	bottom: 2px;
 
 	transform: translateY( 100% );
```
Comment (oleq) [inline theme/components/dropdown.scss]: Same as above.

File: theme/components/stickytoolbar.scss [hunk 6]
```diff
@@ -5,14 +5,12 @@
 	.ck-toolbar.ck-toolbar_sticky {
 		@include ck-drop-shadow();
 
-		position: fixed;
 		top: 0;
```
Comment (oleq) [inline theme/components/stickytoolbar.scss]: Shouldn't it be a part of Wireframe? I'm confused about `top, left, right, bottom` and `transform: translate()`. On one hand, they're very basic definitions of what the component should look like. On the other hand, they could be customized by other themes.

File: theme/components/dropdown.scss (hunk 2 above)
Comment (postJScriptum) [inline theme/components/dropdown.scss]: Yep, I think that should be set some default position and the middle seems to be the best.

File: theme/components/button.scss (hunk 1 above)
Comment (postJScriptum) [inline theme/components/button.scss]: +L Maybe simply set ```display: inline-block``` for buttons in wireframe theme. Float left seems to be more problematic in connection with another element without float value.

File: theme/components/dropdown.scss (hunk 3 above)
Comment (postJScriptum) [inline theme/components/dropdown.scss]: +W From the same reasons as above.

File: theme/components/dropdown.scss (hunk 4 above)
Comment (postJScriptum) [inline theme/components/dropdown.scss]: +W Left position should be set as default.

File: theme/components/dropdown.scss (hunk 5 above)
Comment (postJScriptum) [inline theme/components/dropdown.scss]: +W

File: theme/components/icon.scss [hunk 7]
```diff
@@ -3,22 +3,9 @@
...
-
-	* {
-		// Allows dynamic coloring of the icons.
-		color: inherit;
-		// Needed by Webkit/Blink.
-		fill: currentColor;
-	}
-}
-
 .ck-icon {
```
Comment (postJScriptum) [inline theme/components/icon.scss]: Set of icons is delivered by CKEditor, so I think that this is the Wireframe area. An icon doesn't display without width and height value, those values could be anytime overridden by the extended theme.

File: theme/components/stickytoolbar.scss (hunk 6 above)
Comment (postJScriptum) [inline theme/components/stickytoolbar.scss]: Maybe the best way to decide what is the best place for them is the question: "If this element will look OK without this properties." If not then should be placed in the Wireframe.

**Comment by Reinmar** [comment]
That's how the link sample look like without the Lark theme loaded (with the wireframe only):


image

You can see the balloon there, toolbar layout, etc. So I think the job was done nicely :)

**Pull request merged**